### PR TITLE
feat(template): support package.json and handler

### PR DESCRIPTION
**Pull request by clarkdo**
Add `package.json` and `package.js` for each package.

With this pr, we can support
- Auto deps pr from renovate, then we can keep updating packages to latest version
- Isolate each template package into its own dir
- Easier to add a new framework and better control by user `package.js`

### CODE REVIEW

File: packages/create-nuxt-app/lib/saofile.js [hunk 1]
```diff
@@ -129,17 +132,27 @@ module.exports = {
...
-        delete data.scripts['']
-        delete data.dependencies['']
-        delete data.devDependencies['']
-        return data
+        return { ...data, ...pkg.load(generator) }
       }
     })
 
+    // For compiling pacakge.json
+    actions.push({
```
Comment (Atinux) [inline packages/create-nuxt-app/lib/saofile.js:145]: Typo here

File: packages/cna-template/template/nuxt/package.js [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+      delete pkg.devDependencies['eslint-plugin-nuxt']
+    }
+    if (!lintStaged) {
+      delete pkg.husky
+      delete pkg['lint-staged']
+      delete pkg.devDependencies.husky
+      delete pkg.devDependencies['lint-staged']
+    }
+    if (!stylelint) {
+      lintStaged && delete pkg['lint-staged']['*.{css,vue']
```
Comment (NickBolles) [inline packages/cna-template/template/nuxt/package.js:35]: Is this missing the trailing `}`?

Comment (NickBolles) [reply packages/cna-template/template/nuxt/package.js:35]: @clarkdo

Comment (clarkdo) [reply packages/cna-template/template/nuxt/package.js:35]: Thanks for the pointing.

Comment (clarkdo) [reply packages/cna-template/template/nuxt/package.js:35]: Will be fixed in https://github.com/nuxt/create-nuxt-app/pull/493/

**Pull request merged**